engine/lib/phx/src/system/guid.rs

```diff
@@ -1,19 +1,22 @@
-#[no_mangle]
-pub static mut nextID: u64 = 1;
+use std::sync::atomic::{AtomicU64, Ordering};
 
-#[no_mangle]
-pub unsafe extern "C" fn GUID_Create() -> u64 {
-    let fresh0 = nextID;
-    nextID = nextID.wrapping_add(1);
-    fresh0
-}
+static NEXT_ID: AtomicU64 = AtomicU64::new(1);
 
-#[no_mangle]
-pub unsafe extern "C" fn GUID_Exists(id: u64) -> bool {
-    id < nextID && id != 0
+#[repr(C)]
+pub struct GUID {
 }
 
-#[no_mangle]
-pub unsafe extern "C" fn GUID_Reset() {
-    nextID = 1;
-}
+#[luajit_ffi_gen::luajit_ffi]
```

Comment:
We usually add some documentation to the code.
Also if you put documentation for methods in the `impl` block with `#[luajit_ffi_gen::luajit_ffi]` it will be put in the generated Lua meta file so script developers can see them with Lua LSP.
Though in this case documentation is not really needed.